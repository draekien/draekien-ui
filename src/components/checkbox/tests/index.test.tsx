/** @jsx jsx */
import { fireEvent, getByText, render } from '@testing-library/react';
import { jsx } from 'theme-ui';
import Checkbox from '..';

describe('Checkbox component', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <Checkbox id="checkbox" label="checkbox" tooltip="tooltip">
        test checkbox
      </Checkbox>
    );

    expect(container).toMatchSnapshot();
  });

  test('should call onChange', (done) => {
    const fn = jest.fn(() => done());
    const component = render(
      <Checkbox id="checkbox" onChange={fn}>
        test
      </Checkbox>
    );

    fireEvent(
      getByText(component.container, 'test'),
      new MouseEvent('click', { bubbles: true, cancelable: true })
    );
  });

  test('should call onToggle', (done) => {
    const fn = jest.fn(() => done());
    const component = render(
      <Checkbox id="checkbox" onToggle={fn}>
        test
      </Checkbox>
    );

    fireEvent(
      getByText(component.container, 'test'),
      new MouseEvent('click', { bubbles: true, cancelable: true })
    );
  });
});
