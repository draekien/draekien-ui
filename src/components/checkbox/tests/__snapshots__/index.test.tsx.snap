// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should match snapshot 1`] = `
<div>
  <div
    class="css-hfz0wm-InputWrapper"
  >
    <label
      class="css-7ojqvb-InputWrapper"
      for="checkbox"
    >
      <div
        class="css-3vy6se-Text"
      >
        checkbox
      </div>
    </label>
    <div
      class="css-1977fki-Checkbox"
    >
      <label
        class="css-lhofcl-Checkbox"
        for="checkbox"
      >
        test checkbox
      </label>
      <div
        class="css-8qt76p-Checkbox"
      >
        <div
          class="css-1mrog33-Checkbox"
        />
      </div>
      <input
        class="css-1bi88cn-Checkbox"
        id="checkbox"
        readonly=""
        type="checkbox"
      />
    </div>
    <label
      class="css-7ojqvb-InputWrapper"
      for="checkbox"
    >
      <div
        class="css-3vy6se-Text"
      >
        tooltip
      </div>
    </label>
  </div>
</div>
`;
